### DISCUSSION TITLE
iOS app makes reading long issues chronologically very hard.

### DISCUSSION THREAD

**Discussion by the-Bruce** [community/community#60625] [Mobile]
### Select Topic Area

Product Feedback

### Body

On the iOS GitHub app, when reading issues with large amounts of activity, the app only loads the initial issue, and the final few comments of the discussion. It then lets you load additional comments in small batches from the bottom upwards.

This is obviously fine for people wanting to know what happened and how it was ultimately resolved, but for more complex issues where the context of the discussion is relevant, it is often useful to read the whole thing from start to finish. The present behaviour of the app makes this very frustrating, as not only does it only load comments in weirdly small batches, it then scrolls the load more button out of sight as it attempts to put you back to the last comment previously loaded, assuming you are continuing to read from the bottom to the top. This forces a weird battle between you scrolling up, and the app scrolling you back down again for each new batch of comments.

What would be nice is some way of reducing the frustration of this. My ideal would be an option to load comments top down rather than bottom up, but this may be undesirable from a UI clutter perspective, given I presume I'm in the relative minority in wanting this.
Alternatively, some slightly more subtle logic about where I'm currently scrolled would potentially provide a more natural and less invasive solution, (e.g. if my current scroll position puts the load more comments button in the bottom 5th of the screen, it's unlikely I'm going to want you to scroll me back down again).
Another option would be to keep the load more button visible after the scroll (such as making it stick to the top of the pane), which would also alleviate this problem.

I realise this is a kinda niche use-case, but it would be nice to see if anyone else had run into this or had any other thoughts about how it could be addressed.

(Tested on v1.120.0 on iPadOS 16.1.1 on an iPad Air 3rd gen)

**Comment by anhlam2234** [comment]
😊😊

**Comment by taki-on** [comment]
Thank you for sharing the issue and your suggestions. The current UX assumes that the latest timeline events will be the most interesting ones for users, but it is true that there will be some cases when reading from top to bottom would be more useful. I will pass this to our designers and see what we can do.
